### DISCUSSION TITLE
Remove free() from type output?

### DISCUSSION THREAD

**Discussion by m1212e** [wasm-bindgen/wasm-bindgen#3423] [Q&A] [answered]
I'm using two structs as classes whose types are auto generated by wasm_bindgen. They include the `free()` method. If I derive from these classes, the method 'leaks' through to types the end user of my library actually needs to use. Since they will never need to call `free()` manually the methods are a little annoying and pollute my otherwise simple and clean API. I know that the methods need to exist and don't want to actually remove them, but is there any option to just not include them into the TypeScript file? Just like the `#[wasm_bindgen(skip_typescript)]` annotation works?

**Comment by daxpedda** [comment]
See https://rustwasm.github.io/docs/wasm-bindgen/reference/attributes/on-rust-exports/typescript_custom_section.html?highlight=typescript#typescript_custom_section, with that you can define whatever interface you like.

**Reply by m1212e** [reply]
If I understand this correctly, I would need to overwrite the type the struct has completely? So instead of just removing the `free()` method, I would add all the other things?

**Reply by daxpedda** [reply, marked as answer]
As far as I understand: unfortunately yes.

**Reply by m1212e** [reply]
Ok thank you. That would ofcourse kill a lot of the smooth workflow the auto generated types offer. I'm unfamiliar with the codebase and project but how likely would it be that you accept a pull which offers some way of configuring the output of the `free()` method in the types?

**Reply by daxpedda** [reply]
Unfortunately anything related to TypeScript isn't something I'm knowledgeable enough about to review. But other maintainers might find time for that.
> That would ofcourse kill a lot of the smooth workflow the auto generated types offer.

As proposed by the guide, you could probably write your own derive macro to take care of that for you.
